sources/api/storewolf/src/main.rs

```diff
@@ -186,11 +186,50 @@ fn parse_metadata_toml(md_toml_val: toml::Value) -> Result<Vec<model::Metadata>>
             // A table means there is more processing to do. Add the current
             // key and value to the Vec to be processed further.
             toml::Value::Table(table) => {
-                for (key, val) in table {
-                    trace!("Found table for key '{}'", &key);
-                    let mut path = path.clone();
-                    path.push(key.to_string());
-                    to_process.push((path, val));
+                // This is special case to handle metadata as table that contains
+                // "command": "command",
+                // "strength": "weak",
+                // "skip-if-populated": true
+                if table.contains_key("command")
+                    && table.contains_key("strength")
+                    && table.contains_key("skip-if-populated")
```

Comment:
Spoke with @vyaghras in person about this. CC @bcressey as well so we can reconcile opinions here.

To recap: the problem we are solving is that when `storewolf` encounters a TOML table, it recursively descends into the settings or metadata within, treating all other TOML value types as "leaves" which will be individually stored as metadata. With the changes to `setting-generators`, we wish to represent them as objects and so need `storewolf` to be able to distinguish between "leaf" objects that should be stored, and "branch" objects into which it should recurse.

The proposed solutions that I'm aware of aree:
* Add a new key prefix `metadata-table` for `storewolf`, indicating that the TOML table should be treated as a leaf.
  * Pros: Unambiguous, Generic
  * Cons: Poor clarity: `metadata` and `settings` both align exactly with datastore classifications. Adding `metadata-table` is less clear.
* (Implemented here) search through TOML tables for keys that distinguish specific metadata tables as being `setting-generators`, then store those as objects.
  * Pros: Preserves datastore semantic clarity
  * Cons: Unambiguous, but only if you know the rules on what keys are special. Future metadata objects would need special consideration

Some additional ideas I discussed with @vyaghras were:
* Mark leaf objects in the config with a special key, like `is-object: true`. Storewolf can discard this key after using it to identify the object as a leaf.
  * Pros: Unambiguous, Generic, Maintains datastore semantic clarity
  * Cons: `is-object` would live in the keyspace that is otherwise reserved for settings data.
* Add a postfix or suffix to the TOML table key which is otherwise "illegal" in the keyspace which indicates that a table is a leaf. e.g. `[metadata.top-level.sub-level~]` or `[~metadata.top-level.sub-level]`.
  * Pros: Unambiguous, Generic, maintains datastore semantic clarity
  * Cons: `~` is somewhat opaque.

For what it's worth, I'm partial to the last suggestion because it keeps the keyspace open everywhere else for the data being stored, but curious what ya'll think. I'm also open to a suggestion that I'm overcomplicating this :laughing: